src/util/check-network.ts

```diff
@@ -0,0 +1,16 @@
+export default async function checkOnline(): Promise<boolean> {
+    let online = true
+    await fetch('http://www.google.com', {
```

Comment:
This is not a reliable way to check this. It assumes access to `google.com` is a reliable way to consider a user has access to the Internet, although that rules out lots of situations including a huge chunk of our users in the PRC (https://en.wikipedia.org/wiki/Websites_blocked_in_mainland_China). It may however be a reliable way to check if we're limiting the use of this module just to the part of the codebase that is built around the Google Drive backup implementation (as you need access to google to use that feature, anyway). So if you move this module to somewhere in there.

We've actually encountered this same issue before, although I think we ended up getting distracted by other stuff and forgetting about it. One thing I learnt at that time was the [`Navigator.onLine` Web API](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine). Though this is not entirely reliable, and doesn't factor in access to the Internet, but it means you don't have to make the user do an unintentional network request.

I would probably go for the first choice for now, although update this function to replace the `.then`, `.catch` with just a standard `try...catch` to do the same thing